sys/net/gnrc/pktbuf_static/gnrc_pktbuf_static.c

```diff
@@ -412,6 +436,18 @@ static void *_pktbuf_alloc(size_t size)
         max_byte_count = last_byte;
     }
 #endif
+
+    const void *mismatch;
+    if (CONFIG_GNRC_PKTBUF_CHECK_USE_AFTER_FREE &&
+        (mismatch = mem_is_set(ptr + 1, CANARY, size - sizeof(_unused_t)))) {
+        printf("[%p] mismatch at offset %u/%u (ignoring %d initial bytes that were repurposed)\n",
+               (void *)ptr, (uintptr_t)mismatch - (uintptr_t)ptr, size, sizeof(_unused_t));
```

Comment:
```suggestion
        printf("[%p] mismatch at offset %" PRIxPTR "/%zu (ignoring %zu initial bytes that were repurposed)\n",
               (void *)ptr, (uintptr_t)mismatch - (uintptr_t)ptr, size, sizeof(_unused_t));
```

`"%"PRIuPTR` can be used to print `uintptr_t`, `"%zu"` for `size_t`.

Reply:
I'm pretty sure there is no `%z` on newlib

Reply:
Hmm, maybe casting to `unsigned` instead while staying with `%u` to please the CI?

`PRIuPTR` actually maps to something non-magical and should even work with a limited subset of supported printf format specifiers.